app/bingey-ui/src/components/WatchlistsControler/WatchlistAdd/WatchlistAdd.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  TextField,
+} from '@material-ui/core';
+import Slide from '@material-ui/core/Slide';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogUI from '@material-ui/core/Dialog';
+import './WatchlistAdd.scss';
+import AddCircleIcon from '@material-ui/icons/AddCircle';
+
+const Transition = React.forwardRef(function Transition(props, ref) {
+  return <Slide direction='up' ref={ref} {...props} />;
+});
+
+export function WatchlistAdd({ onCreateWatchlist, openDrawer }) {
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [watchlistName, setWatchlistName] = useState('');
+
+  const closeDialog = () => {
+    setDialogOpen(false);
+  };
+
+  const openDialog = () => {
+    openDrawer();
+    setDialogOpen(true);
+  };
+
+  const handleInputChange = (e) => {
+    setWatchlistName(e.target.value);
+  };
+
+  return (
+    <>
+      <ListItem button onClick={openDialog} key={'create'}>
+        <ListItemIcon>
+          <AddCircleIcon />
+        </ListItemIcon>
+        <ListItemText primary={'New Watchlist'} />
+      </ListItem>
+      <DialogUI
```

Comment:
I think this should be moved to a separate file as it's own component. It's already rather large and could potentially grow in the future

Reply:
Created a separate Basic Dialog component that should be used where a basic dialog is needed. This reduces the code quite a bit so I think this can stay like this